cl/citations/management/commands/westlaw_citation_merger.py

```diff
@@ -0,0 +1,354 @@
+    citations = prepare_citation(citation)
+
+    if not citations:
+        logger.warning(
+            f"The citation: {citation} you are trying to add to the cluster "
+            f"id: {cluster_id} is invalid."
+        )
+        return
+
+    # Get correct reporter type before trying to add the citation
+    if not citations[0].corrected_reporter():
+        reporter_type = Citation.STATE
+    else:
+        cite_type_str = citations[0].all_editions[0].reporter.cite_type
+        reporter_type = map_reporter_db_cite_type(cite_type_str)
+
+    try:
+        if not debug:
+            Citation.objects.get_or_create(
+                volume=citations[0].groups["volume"],
+                reporter=citations[0].corrected_reporter(),
+                page=citations[0].groups["page"],
+                type=reporter_type,
+                cluster_id=cluster_id,
+            )
+        logger.info(f'Citation "{citation}" added to cluster id: {cluster_id}')
+    except IntegrityError:
+        logger.warning(
+            f"Reporter mismatch for cluster: {cluster_id} on cite: {citation}"
+        )
+
+
+def prepare_date(date_str: str) -> Optional[datetime]:
+    """Convert dates like 'February 28, 2011' or '2011-02-28' to date object
+    :param date_str: date string
+    :return: date object or None
+    """
+    valid_formats = ["%B %d, %Y", "%Y-%m-%d"]
+    for date_format in valid_formats:
+        try:
+            date_obj = datetime.strptime(date_str, date_format)
+            return date_obj
+        except ValueError:
+            logger.warning(f"Invalid date string: {date_str}")
+            continue
+    return None
+
+
+def prepare_citation(citation: str) -> Union[List[FullCaseCitation], List]:
+    """Convert str citation to valid citation objects
+    :param citation: citation str
+    :return: list of valid cites
+    """
+    clean_cite = re.sub(r"\s+", " ", citation)
+    cites = get_citations(clean_cite)
+    cites = [cite for cite in cites if isinstance(cite, FullCaseCitation)]
+    return cites
+
+
+def process_citations(data: DataFrame | TextFileReader, debug: bool) -> None:
+    """
+    Process citations from csv file
+    :param data: rows from csv file
+    :param debug: if true don't save changes
+    :return: None
+    """
+    for index, row in data.iterrows():
+        case_name = row.get("Title")
+        citation = row.get("Citation")
+        parallel_citation = row.get("Parallel Cite")
+        court = row.get("Court Line")
+        docket_number = row.get("Docket Num")
+        date_filed = row.get("Filed Date")
+
+        additional_search_params = {}
+
+        if court:
+            # Add court to search params if exists
+            if court.endswith("."):
+                # Remove dot at end, court-db fails to find court with
+                # dot at end, e.g. "Supreme Court of Pennsylvania." fails,
+                # but "Supreme Court of Pennsylvania" doesn't
+                court = court[:-1]
+
```

Comment:
why not just strip `.` from court instead of checking ?

Reply:
you are right, it's easier that way